cdap-data-fabric/src/main/java/co/cask/cdap/data2/datafabric/dataset/service/DatasetInstanceService.java

```diff
@@ -179,9 +229,11 @@ private DatasetMeta getFromMds(Id.DatasetInstance instance) throws Exception {
    * @throws DatasetTypeNotFoundException if the dataset type was not found
    * @throws Exception if something went wrong
    */
-  public void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
+  void create(String namespaceId, String name, DatasetInstanceConfiguration props) throws Exception {
     // Throws NamespaceNotFoundException if the namespace does not exist
     Id.Namespace namespace = ConversionHelpers.toNamespaceId(namespaceId);
+    Principal principal = SecurityRequestContext.toPrincipal();
```

Comment:
Can DatasetInstanceService serve anything other than the dataset instance REST APIs? For program containers, where we want to use UGI, it goes via DatasetFramework/DatasetServiceClient right? whatever the DatasetServiceClient sets in the request header appears here in SecurityRequestContext. But we can change it in my next PR to always use an injected AuthenticationContext, which has an HTTP implementation or a UGI implementation.